packages/explorerkit-server/src/server.ts

```diff
@@ -203,67 +88,91 @@ app.post("/decode/accounts", responseDurationMiddleware, async (req: Request, re
   }
 });
 
+const decodeErrorsSchema = z.object({
+  errors: z.array(
+    z
+      .object({
+        programId: z.string(),
+        errorCode: z.coerce.number().nullable().optional(),
+      })
+      .nullable()
+  ),
+});
+
+// Endpoint to decode program errors
+app.post("/decode/errors", responseDurationMiddleware, async (req: Request, res: Response) => {
+  const { data, error } = decodeErrorsSchema.safeParse(req.body);
+
+  if (error) {
+    return res.status(400).json({ error: "Invalid request body", errors: error.errors });
+  }
+
+  const programIdsWithFailure = data.errors
+    .filter((err) => err?.errorCode)
+    .map((err) => err?.programId)
+    .map((v) => v!);
+
+  const idls = await loadAllIdls(programIdsWithFailure);
+  const decodedErrors = data.errors.map((error) => error && decodeProgramError(idls, error));
```

Comment:
SUPER NIT: can you type the response structure? Renaming this variable would cause a breaking change